Happy Dance/Assets/Fader/Fader.cs

```diff
@@ -0,0 +1,22 @@
+﻿using UnityEngine;
+using UnityEngine.Assertions;
+
+/// <summary>
+/// Fader is used to controller the color of a target material. It is driven by an animation clip.
+/// </summary>
+public class Fader 
+    : MonoBehaviour
+{
+    public Color Color;
+    public Material FaderMaterial;
+
+    private void OnValidate()
+    {
+        Assert.IsNotNull(FaderMaterial, "FaderMaterial must not be null.");
+    }
+
+    void Update()
```

Comment:
void Update() > private void Update() 

even though Update is already private, it would be more consistent to add the keyword private.

Reply:
Done